editor/code_editor.cpp

```diff
@@ -1425,6 +1436,9 @@ void CodeTextEditor::move_lines_down() {
 
 			int next_start_pos = text_editor->has_selection(c_next) ? text_editor->get_selection_from_line(c_next) : text_editor->get_caret_line(c_next);
 			int next_end_pos = text_editor->has_selection(c_next) ? text_editor->get_selection_to_line(c_next) : text_editor->get_caret_line(c_next);
+			if (text_editor->get_selection_to_column(c_next) == 0 && text_editor->has_selection(c_next)) {
```

Comment:
Should swap these around, and above, else `get_selection_to_column` will error out if there is not a selection.

```suggestion
			if (text_editor->has_selection(c_next) && text_editor->get_selection_to_column(c_next) == 0) {
```

Reply:
Oh, that's right. I remember having them in right order, but due to some brain fart ended up switching them. Thought about readability or something...